clippy_utils/src/qualify_min_const_fn.rs

```diff
@@ -375,10 +375,8 @@ fn is_const_fn(tcx: TyCtxt<'_>, def_id: DefId, msrv: Option<&RustcVersion>) -> b
                         .expect("`rustc_attr::StabilityLevel::Stable::since` is ill-formatted"),
                 )
             } else {
-                // `rustc_mir::const_eval::is_const_fn` should return false for unstably const functions.
-                unreachable!();
+                // Unstable const fn with the feature enabled.
+                msrv.is_none()
```

Comment:
Why does the return value change here depending on whether MSRV is set?

Reply:
There's no stable rust version where the function is `const` as there's no stability attribute. If the MSRV is set, then the function is clearly not const. If it's not set, then the question becomes is the function `const` now, which is true.